tests/test_csc.py

```diff
@@ -151,6 +156,64 @@ def setUp(self):
             self.assertEqual(alarm.maxSeverity, AlarmSeverity.NONE)
             self.assertFalse(alarm.acknowledged)
 
+    async def test_mute(self):
+        """Test the mute and unmute command."""
+        async with Harness(config_dir=TEST_CONFIG_DIR) as harness:
+            await salobj.set_summary_state(harness.remote, state=salobj.State.ENABLED,
+                                           settingsToApply="enabled.yaml")
+            nrules = len(harness.csc.model.rules)
+
+            user1 = "test_mute 1"
+            # mute all alarms for a short time,
+            # then wait for them to unmute themselves
+            await harness.remote.cmd_mute.set_start(name="Enabled.*",
+                                                    duration=0.1,
+                                                    severity=AlarmSeverity.SERIOUS,
+                                                    mutedBy=user1,
+                                                    timeout=STD_TIMEOUT)
+            muted_names = []
+            while len(muted_names) < nrules:
+                data = await harness.remote.evt_alarm.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(data.mutedSeverity, AlarmSeverity.SERIOUS)
+                self.assertEqual(data.mutedBy, user1)
+                muted_names.append(data.name)
+            self.assertEqual(set(muted_names), set(harness.csc.model.rules))
```

Comment:
Isn't this assert a tautology, given the nature of the while loop above? I don't see how we could exit the while loop and have this assert fail.

Reply:
No. The loop picks up *all* alarm events. They don't have to be triggered by muting or unmuting.

What I am testing is that the first batch of events is for the alarms to be muted. Once those have arrived I wait for more and check that they are unmuted.

I'll add a code comment.

Reply:
Oh, ok, I see the `set` part of the test now. Your addition there helps, too.